### PR TITLE
feat: upgrade to opendal 0.47 in object store

### PR DESCRIPTION
**Pull request by hzxa21**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

- Upgrade to opendal 0.47 in object store
- Adopt the new reader writer API in opendal object store
- support streaming uploader metrics via the new executor trait

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by wcy-fdu** [comment]
this pr's longevity test pass on azure blob.
image

**Comment by hzxa21** [comment]
Split the chrono upgrade related change in #17279

**Comment by wcy-fdu** [comment]
BTW, will we switch to opendal s3 in next pr?

**Comment by hzxa21** [comment]
> BTW, will we switch to opendal s3 in next pr?

Let's do it in a separate PR.

**Review by stdrc — APPROVED** [review]

File: src/object_store/src/object/opendal_engine/opendal_object_store.rs [hunk 1]
```diff
@@ -273,34 +332,64 @@ impl OpendalStreamingUploader {
...
+            .concurrent(config.s3.developer.upload_concurrency)
+            .executor(Executor::with(monitored_execute))
             .await?;
-        Ok(Self { writer })
+        Ok(Self {
+            writer,
+            buf: vec![],
+            not_uploaded_len: 0,
+            is_valid: true,
+            abort_on_err: config.s3.developer.writer_abort_on_err,
```
[the remark this answers is not in the document]
Comment (hzxa21) [reply src/object_store/src/object/opendal_engine/opendal_object_store.rs]: Good catch. Fixed.

**Pull request merged**